lib/observer/src/observer_port_wx.erl

```diff
@@ -267,10 +267,18 @@ handle_cast(Event, _State) ->
     error({unhandled_cast, Event}).
 
 handle_info({portinfo_open, PortIdStr},
-	    State = #state{grid=Grid, ports=Ports, open_wins=Opened}) ->
-    Port = lists:keyfind(PortIdStr,#port.id_str,Ports),
-    NewOpened = display_port_info(Grid, Port, Opened),
-    {noreply, State#state{open_wins = NewOpened}};
+	    State = #state{node=Node, grid=Grid, opt=Opt, open_wins=Opened}) ->
+    Ports0 = get_ports(Node),
+    Ports = update_grid(Grid, Opt, Ports0),
+    Port = lists:keyfind(PortIdStr, #port.id_str, Ports),
+    NewOpened =
+        case Port of
+            undefined ->
```

Comment:
lists:keyfind/3 returns `false` and not `undefined` when the key is not found, so this clause will never match.